### PR TITLE
Add TOKEN-PERCENTAGE-PURCHASE.md

### PR DESCRIPTION
**Pull request by TJ-Everett**
Refactor TOKEN-PERCENTAGE-SELL.md with updated formatting Update RULE-GUIDE.md with purchase + sell rule

### CODE REVIEW

File: docs/userGuides/rules/TOKEN-PERCENTAGE-PURCHASE.md [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+                uint256 currentTotalSupply, 
+                uint256 amountToTransfer, 
+                uint64 lastPurchaseTime, 
+                uint256 purchasedWithinPeriod
+            ) 
+            external 
+            view 
+            returns (uint256);
+        ```
+- in Asset Handler:
```
Comment (oscarsernarosero) [inline docs/userGuides/rules/TOKEN-PERCENTAGE-PURCHASE.md]: we should add a link to the handler.

**Review by oscarsernarosero — APPROVED** [review]
It looks really good Tayler. I just have a small suggestion, and current implementation does not bypass admins, but let's wait until we clear that out. So Im approving.

File: docs/userGuides/rules/TOKEN-PERCENTAGE-PURCHASE.md (hunk 1 above)
Comment (TJ-Everett) [reply docs/userGuides/rules/TOKEN-PERCENTAGE-PURCHASE.md]: Great call added to purchase and sell percentage docs

**Comment by TJ-Everett** [comment]
> It looks really good Tayler. I just have a small suggestion, and current implementation does not bypass admins, but let's wait until we clear that out. So Im approving.

Thanks Oscar, I did remove from these since part of the ticket to fix that in the internal branch should be to update the docs of all the tickets that need it, if we decide to skip those for admins.

File: docs/userGuides/rules/TOKEN-PERCENTAGE-PURCHASE.md [hunk 2]
```diff
@@ -0,0 +1,223 @@
...
+
+This rule works at the AMM level. It must be activated and configured for each desired AMM in the corresponding AMM handler. This rule will not be applied at the token level and will only be checked through the AMM swap function. 
+
+## Data Structure
+
+A token-percentage-purchase rule is composed of 4 components:
+
+- **Token Percentage** (uint16): The maximum percent in basis units of total supply able to be purchased during the *period* (i.e. 5050 = 50.50% of total supply). 
+- **Purchase  Period** (uint16): The length of time for which the rule will apply, in hours.
+- **Starting Timestamp** (uint64): The timestamp of the date when the *period* starts counting.
```
Comment (ShaneDuncan602) [inline docs/userGuides/rules/TOKEN-PERCENTAGE-PURCHASE.md]: I think we are describing all of these as Unix timestamps.

**Review by ShaneDuncan602 — APPROVED** [review]
One tiny comment but I'm approving. Looks really good.

**Pull request merged**